### PR TITLE
Assign a random team member on r? @rust-lang/<team>

### LINKED ISSUE #187: Assign a random team member with r+ priviliges on r? @rust-lang/<team>
If someone writes:
```
r? @rust-lang/docs 
```
Then we could for example assign steveklabnik since they have r+ rights.

This mistake happens from time to time; what happens now when someone writes the above is to leave the PR without a reviewer...

cc https://github.com/rust-lang/rust/pull/57357#issuecomment-451675541

### PR DESCRIPTION
**Pull request by LeSeulArtichaut**
Attemps to close #187.

### CODE REVIEW

File: highfive/newpr.py
Comment (jyn514) [inline highfive/newpr.py:39]: We only want to match rust-lang, don't we?
```suggestion
reviewer_re = re.compile("\\b[rR]\?[:\- ]*@(rust-lang/)?([a-zA-Z0-9\-]+)")
```

**Comment by camelid** [comment]
@LeSeulArtichaut this would be great to have! Are you planning on getting back to this?

**Comment by LeSeulArtichaut** [comment]
This depends on whether @Mark-Simulacrum wants to proceed with https://github.com/rust-lang/triagebot/pull/433 first, but as that PR encountered some difficulties, maybe this can be revisited?

**Comment by Mark-Simulacrum** [comment]
Yeah, this seems like the right approach in the interim while we wait for GitHub to resolve the problems preventing full rollout of r? for assignment in triagebot. From a quick scroll through it looks like this wants to add team groups for the existing teams (probably at least libs and compiler perhaps?) to the real files too? I guess we can do that in follow-up PRs.

File: highfive/newpr.py
Comment (LeSeulArtichaut) [reply highfive/newpr.py:39]: I think we have also configs for rust-embedded, rust-nursery and rust-deprecated.

**Comment by LeSeulArtichaut** [comment]
*This code looks really bad, I wonder who wrote it... Oh, hey me from (exactly) 9 months ago!*

> From a quick scroll through it looks like this wants to add team groups for the existing teams (probably at least libs and compiler perhaps?) to the real files too? I guess we can do that in follow-up PRs.

If I understand the code correctly, it first checks if a group matches the team name, so in `r? @rust-lang/libs` it's going to check if `libs` exists :
```python
groups.get(match.group(2))
```
before checking if `rust-lang/libs` exists :
```python
potential = groups.get("%s/%s" % (match.group(1), match.group(2)))
```
So I think we don't need to change the data files, because these teams already exist in https://github.com/rust-lang/highfive/blob/f7434f5b191f65b08c0faf35301f0edc585f75dd/highfive/configs/rust-lang/rust.json#L2-L12

**Comment by wesleywiser** [comment]
I would **love** to see this feature added. Is there anything I can do to help get this merged? I see this needs a rebase and maybe another review?

**Comment by LeSeulArtichaut** [comment]
Rebased.

File: highfive/newpr.py
Comment (wesleywiser) [inline highfive/newpr.py:252]: Do we need an `else: return None` to go with this `if`? Sorry, I don't know anything about Python 😄

File: highfive/newpr.py
Comment (wesleywiser) [inline highfive/newpr.py:262]: Does this mean `r? @rust-lang/compiler` will potentially pick someone outside that group? (I'm trying to figure out why we want this branch)

File: highfive/newpr.py
Comment (LeSeulArtichaut) [reply highfive/newpr.py:252]: No, if we don't `return` Python considers we're returning `None` implicitely

File: highfive/newpr.py
Comment (LeSeulArtichaut) [reply highfive/newpr.py:262]: I *think* I intended this for repos outside of `rust-lang/rust`, e.g. embedded folks. In `rust-lang/rust` the `all` team.

Comment (wesleywiser) [reply highfive/newpr.py:262]: Oh I see 

https://github.com/rust-lang/highfive/blob/ab9ad47a683b488de3d85c69445c8daced06c8df/highfive/configs/rust-lang/rust.json#L3

**Review by wesleywiser — APPROVED** [review]
The changes seem reasonable to me but I don't know anything about Python

**Comment by wesleywiser** [comment]
@Mark-Simulacrum I think this is good to go if you want to take a look. 🙂

**Pull request merged**
